worlds/sm64ex/Options.py

```diff
@@ -114,35 +115,39 @@ class StrictMoveRequirements(DefaultOnToggle):
     if Move Randomization is enabled"""
     display_name = "Strict Move Requirements"
 
-def getMoveRandomizerOption(action: str):
-    class MoveRandomizerOption(Toggle):
-        """Mario is unable to perform this action until a corresponding item is picked up.
-        This option is incompatible with builds using a 'nomoverando' branch."""
-        display_name = f"Randomize {action}"
-    return MoveRandomizerOption
-
-
-sm64_options: typing.Dict[str, type(Option)] = {
-    "AreaRandomizer": AreaRandomizer,
-    "BuddyChecks": BuddyChecks,
-    "ExclamationBoxes": ExclamationBoxes,
-    "ProgressiveKeys": ProgressiveKeys,
-    "EnableCoinStars": EnableCoinStars,
-    "StrictCapRequirements": StrictCapRequirements,
-    "StrictCannonRequirements": StrictCannonRequirements,
-    "StrictMoveRequirements": StrictMoveRequirements,
-    "AmountOfStars": AmountOfStars,
-    "FirstBowserStarDoorCost": FirstBowserStarDoorCost,
-    "BasementStarDoorCost": BasementStarDoorCost,
-    "SecondFloorStarDoorCost": SecondFloorStarDoorCost,
-    "MIPS1Cost": MIPS1Cost,
-    "MIPS2Cost": MIPS2Cost,
-    "StarsToFinish": StarsToFinish,
-    "death_link": DeathLink,
-    "CompletionType": CompletionType,
-}
+class MoveRandomizerOption(Toggle):
+    """Mario is unable to perform this action until a corresponding item is picked up.
+    This option is incompatible with builds using a 'nomoverando' branch."""
+
+for action in action_item_table:
+    move_option_class_name = f"MoveRandoOption_{action.replace(' ','')}"
+    vars()[move_option_class_name] = type(move_option_class_name, MoveRandomizerOption.__bases__, dict(MoveRandomizerOption.__dict__))
+    vars()[move_option_class_name].display_name = f"Randomize {action}"
```

Comment:
Shouldn't this be with the other loop?

Reply:
Idea is to create all classes before the definition of dataclass_members, so this is just a readability thing.